include/irr/asset/IAsset.h

```diff
@@ -167,25 +173,40 @@ class IAsset : virtual public core::IReferenceCounted
 
         virtual core::smart_refctd_ptr<IAsset> clone(uint32_t _depth = ~0u) const = 0;
 
+		inline E_MUTABILITY getMutability() const { return m_mutability; }
+		inline bool isMutable() const { return getMutability() == EM_MUTABLE; }
+		inline bool canBeConvertedToDummy() const { return (getMutability()&EM_CPU_PERSISTENT) != EM_CPU_PERSISTENT; }
```

Comment:
not sure I want these as public.

We should have a `setMutability`

Reply:
setting asset's mutability is exclusive for asset manager and it's already a friend of IAsset

Reply:
ah ok, but still the public functions here are a bit ugly, they should be protected.

Reply:
shouldnt user be able to query asset's mutability level?

Reply:
yes, but we shouldn't have multiple functions to query, just `getMutability` and the enum

obvs `canBeConvertedToDummy` can stay, cause it does more than checking an enum

Reply:
ok `isMutable` might actually be useful